backend/src/main/java/coffeeshout/global/redis/RedisStreamEventRegistrar.java

```diff
@@ -34,15 +32,55 @@ public void afterSingletonsInstantiated() {
     public void inspectConsumerBeans() {
         final String[] beanNames = beanFactory.getBeanNamesForType(Consumer.class);
         for (String beanName : beanNames) {
-            if (beanFactory.getBeanDefinition(beanName) instanceof RootBeanDefinition) {
-                final ResolvableType type = beanFactory.getMergedBeanDefinition(beanName).getResolvableType();
-                final Class<?>[] classes = type.resolveGenerics();
-                if (classes.length == 0 || classes[0] == null) {
-                    continue;
-                }
-                redisObjectMapper.registerSubtypes(classes[0]);
-                log.debug("Registered event subtype: {} from bean: {}", classes[0].getSimpleName(), beanName);
+            try {
+                registerEventTypeFromConsumer(beanName);
+            } catch (Exception e) {
+                log.warn("Failed to process Consumer bean: {} - {}", beanName, e.getMessage(), e);
             }
         }
     }
+
+    private void registerEventTypeFromConsumer(String beanName) {
+        final Class<?> eventType = resolveEventType(beanName);
+        if (eventType == null) {
+            return;
+        }
+
+        if (!BaseEvent.class.isAssignableFrom(eventType)) {
+            log.debug("Skipping bean {} - generic type {} is not a BaseEvent subtype",
+                    beanName, eventType.getSimpleName());
+            return;
+        }
+
+        redisObjectMapper.registerSubtypes(eventType);
+        log.debug("Registered event subtype: {} from bean: {}", eventType.getSimpleName(), beanName);
+    }
+
+    private Class<?> resolveEventType(String beanName) {
+        final ResolvableType type = beanFactory.getMergedBeanDefinition(beanName).getResolvableType();
+        if (type == ResolvableType.NONE) {
+            log.warn("Cannot resolve type for bean: {}", beanName);
+            return null;
+        }
+
+        final ResolvableType consumerType = type.as(Consumer.class);
+        if (consumerType == ResolvableType.NONE) {
+            log.warn("Bean {} does not properly implement Consumer interface", beanName);
+            return null;
+        }
+
+        final ResolvableType genericType = consumerType.getGeneric(0);
+        if (genericType == ResolvableType.NONE) {
+            log.warn("Bean {} has raw type Consumer without generic parameter", beanName);
+            return null;
+        }
+
+        final Class<?> eventType = genericType.resolve();
+        if (eventType == null) {
+            log.warn("Cannot resolve generic type for bean: {}", beanName);
+            return null;
+        }
```

Comment:
`resolve`는 `ResolvableType`이 NONE이거나 없으면 null을 반환해서 아래의 코드로 간소화 할 수 있을 것 같습니다!
```suggestion
    return type.as(Consumer.class).getGeneric(0).resolve();
```